api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -92,13 +93,10 @@ public AppFrameworkServiceImpl(AllAppTemplates allAppTemplates, AllAppDescriptor
 		this.locationService = locationService;
 		this.featureToggles = featureToggles;
 		this.appFrameworkConfig = appFrameworkConfig;
-		this.javascriptEngine = new ScriptEngineManager().getEngineByName("JavaScript");
+		System.setProperty("polyglot.js.nashorn-compat", "true");
+		this.javascriptEngine = new ScriptEngineManager().getEngineByName("graal.js");
```

Comment:
Do we need line 97 above? Shouldn't line 95 that you removed automatically take care of that?

Reply:
Yes, I updated it.